### ISSUE TITLE
InputGroup - remove automated aria-describedby attribute

### ISSUE THREAD

**Issue by thatblindgeye** [patternfly/patternfly-react#9667]
### Discussed in https://github.com/orgs/patternfly/discussions/5954

<div type='discussions-op-text'>

<sup>Originally posted by **dgdavid** September 20, 2023</sup>
Hi there!

Excuse me for the noob question, but I do not get why the `InputGroup` component automatically set the `aria-describedby` attribute to the _form-control_ child pointing to another child component [when the former does not have such an attribute and the latest isn't considered a _form-control_ but has set the `id` attribute](https://github.com/patternfly/patternfly-react/blob/cfd927cf1b7c9315f5ab244368a909f4893464d2/packages/react-core/src/components/InputGroup/InputGroup.tsx#L52-L73).

Despite reading the code and searching on the Github repository, I haven't found anything more than [evidence that it was done on purpose](https://github.com/patternfly/patternfly-react/blob/e27aac2f35e0a8bb16fa5397eb5ef841b3ecc578/packages/react-core/src/components/InputGroup/__tests__/InputGroup.test.tsx#L11-L27). Even looking at the PatternFly [examples](https://www.patternfly.org/components/input-group#examples)... the behavior there don't make sense to me.

Consider an `InputGroup` that groups a `TextInput` of type _password_ with a `Button` that toggles it to type _text_. Due to requirements, the button has an `id` attribute. I cannot see the value of the `TextInput` having set the `aria-describedby="button-id"` attribute. Most probably, I'm missing something.

---

```jsx
import EyeIcon from '@patternfly/react-icons/dist/esm/icons/eye-icon';
import { Button, InputGroup, InputGroupItem, TextInput } from '@patternfly/react-core';

export default function Example() {
  return (
    <InputGroup>
      <InputGroupItem isFill>
        <TextInput id="password" type="password" aria-label="User password field" />
      </InputGroupItem>
      <InputGroupItem>
        <Button id="button-id" variant="control" icon={<EyeIcon />} />
      </InputGroupItem>
    </InputGroup>
  );
}
```
![Screenshot from 2023-09-20 23-09-37](https://github.com/patternfly/patternfly-react/assets/1691872/937e321d-a7fd-4281-9efd-dfdf70d1166e)

---

Looking at the code, it's clear that [the "workaround" is setting the `aria-describedby=""`](https://github.com/patternfly/patternfly-react/blob/e27aac2f35e0a8bb16fa5397eb5ef841b3ecc578/packages/react-core/src/components/InputGroup/InputGroup.tsx#L59) to the input field... but it feels weird. Wouldn't it make better sense the other way around? I.e., the developer setting the `aria-describedby="button-id"` if really needed.

I really appreciate your help to understand the rationale of that behavior.

Thanks a lot in advance.

</div>

**Comment by wise-king-sullyman** [comment]
@dgdavid if you leave a comment here I'll be able to assign you to it since you've already got a PR up.

**Comment by dgdavid** [comment]
> @dgdavid if you leave a comment here I'll be able to assign you to this issue since you've already got a PR up.

Sure, but bear in mind that I sent the PR only for one of the two tasks @thatblindgeye mentioned in the discussion. I fear I cannot help right now with the one for updating the examples.
